servers/graphql-kotlin-spring-server/src/main/kotlin/com/expediagroup/graphql/server/spring/GraphQLExecutionConfiguration.kt

```diff
@@ -42,7 +42,7 @@ class GraphQLExecutionConfiguration {
     @Bean
     @ConditionalOnMissingBean
     fun dataFetcherFactoryProvider(applicationContext: ApplicationContext): KotlinDataFetcherFactoryProvider =
-        SpringKotlinDataFetcherFactoryProvider(applicationContext)
+        SpringSingletonKotlinDataFetcherFactoryProvider(applicationContext)
```

Comment:
by default use `SpringSingletonKotlinDataFetcherFactoryProvider` if no `KotlinDataFetcherFactoryProvider` bean is in the context.

Reply:
should this be an opt-in instead and become default in future releases?